src/dialects/abstract/query-generator.js

```diff
@@ -2221,7 +2221,21 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
   selectFromTableFragment(options, model, attributes, tables, mainTableAs) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
-    let fragment = `SELECT ${attributes.join(', ')} FROM ${tables}`;
+    let fragment = `SELECT`;
+
+    if (typeof options.maxExecutionTimeMs === 'number' && this.dialect.supports.maxExecutionTimeHint) {
```

Comment:
Our new policy is to always throw if someone uses an option that's not supported by their dialect:

```suggestion
    if (options.maxExecutionTimeMs != null) {
      if (!this.dialect.supports.maxExecutionTimeHint) {
        throw new Error(`The maxExecutionTimeMs option is not supported by ${this.dialect.name}`);
      }
```